modules/assim.sequential/R/sda.enkf_NorthAmerica.R

```diff
@@ -0,0 +1,381 @@
+#' @title sda.enkf_NorthAmerica
+#' @name  sda.enkf_NorthAmerica
+#' @author Michael Dietze, Ann Raiho and Alexis Helgeson \email{dietze@@bu.edu}
+#' 
+#' @param settings  PEcAn settings object
+#' @param obs.mean  Lists of date times named by time points, which contains lists of sites named by site ids, which contains observation means for each state variables of each site for each time point. 
+#' @param obs.cov   Lists of date times named by time points, which contains lists of sites named by site ids, which contains observation covariances for all state variables of each site for each time point. 
+#' @param Q         Process covariance matrix given if there is no data to estimate it.
+#' @param pre_enkf_params Used for passing pre-existing time-series of process error into the current SDA runs to ignore the impact by the differences between process errors.
+#' @param ensemble.samples Pass ensemble.samples from outside to avoid GitHub check issues.
+#' @param control   List of flags controlling the behavior of the SDA. 
+#' `TimeseriesPlot` for post analysis examination; 
+#' `OutlierDetection` decide if we want to execute the outlier detection each time after the model forecasting;
+#' `send_email` contains lists for sending email to report the SDA progress;
+#' `keepNC` decide if we want to keep the NetCDF files inside the out directory;
+#' `forceRun` decide if we want to proceed the Bayesian MCMC sampling without observations;
+#' `MCMC.args` include lists for controling the MCMC sampling process (iteration, nchains, burnin, and nthin.).
+#'
+#’ @details
+#’ Restart mode:  Basic idea is that during a restart (primary case envisioned as an iterative forecast), a new workflow folder is created and the previous forecast for the start_time is copied over. During restart the initial run before the loop is skipped, with the info being populated from the previous run. The function then dives right into the first Analysis, then continues on like normal.
+#' 
+#' @description State Variable Data Assimilation: Ensemble Kalman Filter and Generalized ensemble filter. Check out SDA_control function for more details on the control arguments.
+#' 
+#' @return NONE
+#' @import nimble
```

Comment:
Can we importFrom just the functions we need instead of bringing all of nimble into the namespace?

Reply:
I removed it cause I found there is no place where `nimble` is applied.